### PR TITLE
Introducing the Quark Agent Web Interface

### PR DESCRIPTION
**Pull request by pulorsok**
# Quark Agent - Your AI-powered Android APK Analyst

quark agent demo

With Quark Agent, you can perform analyses using only natural language. It creates Quark Script code following your ideas and adjusts the code promptly as you provide feedback.

# Showcase:

Here’s a demonstration of using Quark Agent to detect the CWE-798 vulnerability in the ovaa.apk file.

### Step 1: Environments Requirements

*   Make sure your Python version is 3.8 or above.

### Step 2: Install Quark Agent

*   Install Quark Agent by running:

```bash
git clone https://github.com/quark-engine/quark-engine.git && cd quark-engine
pip install .[QuarkAgent]
```

### Step 3: Prepare the Detection Rule and the Sample File

```bash
.
├── ...
├── quark                   
    ├── ...           
    ├── agent               # Put rule file and sample file here
    ├── ...                
```

You can download the **rule file** [here](https://github.com/quark-engine/quark-script/blob/main/constructCryptoGraphicKey.json) and the **sample file** [here](https://github.com/oversecured/ovaa).

### Step 4: Add your OpenAI API key

Add your OpenAI API key in `quarkAgentWeb.py`

```python
os.environ["OPENAI_API_KEY"] = 'your-api-key-here'
```

### Step 5: Run Quark Agent

```bash
$ cd quark/agent
$ python3 quarkAgentWeb.py

# You can now chat with Quark Agent in your browser. 
# The default URL is http://127.0.0.1:5000
```

Open a browser and navigate to `127.0.0.1:5000` to start using Quark Agent

See more CWE detections using [quark scripts](https://quark-engine.readthedocs.io/en/latest/quark_script.html) and play them with Quark Agent !

### CODE REVIEW

**Comment by 18z** [comment]
# Quark Agent - Your AI-powered Android APK Analyst

quark agent demo

With Quark Agent, you can perform analyses using only natural language. It creates Quark Script code following your ideas and adjusts the code promptly as you provide feedback.

# Showcase:

Here’s a demonstration of using Quark Agent to detect the CWE-798 vulnerability in the ovaa.apk file.

### Step 1: Environments Requirements

*   Make sure your  Python version is 3.8 or above.

### Step 2: Install Quark Agent

*   Install Quark Agent by running:

```bash
git clone https://github.com/quark-engine/quark-engine.git & cd quark-engine
pip install .[QuarkAgent]
```

### Step 3: Prepare the Detection Rule and the Sample File

```bash
.
├── ...
├── quark                   
    ├── ...           
    ├── agent               # Put rule file and sample file here
    ├── ...                
```

### Step 4: Add your OpenAI API key

Add your OpenAI API key in `quarkAgentWeb.py`

```python
os.environ["OPENAI_API_KEY"] = 'your-api-key-here'
```

### Step 5: Run Quark Agent

```bash
$ python3 quark/agent/quarkAgentWeb.py

# You can now chat with Quark Agent in your browser. 
# The default URL is http://127.0.0.1:5000
```

Open a browser and navigate to `127.0.0.1:5000` to start using Quark Agent

File: README.md [hunk 1]
```diff
@@ -32,93 +32,62 @@
...
 
-### Step 2: Install Quark Engine
+### Step 2: Install Quark Agent
 
-*   Install Quark Engine by running:
+*   Install Quark Agent by running:
 
 ```bash
-$ pip3 install -U quark-engine
+git clone https://github.com/quark-engine/quark-engine.git & cd quark-engine
```
Comment (haeter525) [inline README.md:54]: The `&` in the command should be `&&`.

File: README.md [hunk 2]
```diff
@@ -32,93 +32,62 @@
...
+With Quark Agent, you can perform analyses using only natural language. It creates Quark Script code following your ideas and adjusts the code promptly as you provide feedback.
+
+# Showcase:
+
+Here’s a demonstration of using Quark Agent to detect the CWE-798 vulnerability in the ovaa.apk file.
 
 ### Step 1: Environments Requirements
 
-*   Quark requires Python 3.8 or above.
+*   Make sure your  Python version is 3.8 or above.
```
Comment (haeter525) [inline README.md:47]: There is a extra space between `your` and `Python`.

**Review by haeter525 — CHANGES_REQUESTED** [review]

**Review by haeter525 — APPROVED** [review]
LGTM. Thank @pulorsok.

**Pull request merged**